Content/Library/build.fsx

```diff
@@ -208,26 +222,26 @@ Target.create "WatchTests" <| fun _ ->
     let cancelEvent = Console.CancelKeyPress |> Async.AwaitEvent |> Async.RunSynchronously
     cancelEvent.Cancel <- true
 
-let (|Fsproj|Csproj|Vbproj|) (projFileName:string) =
-    match projFileName with
-    | f when f.EndsWith("fsproj") -> Fsproj
-    | f when f.EndsWith("csproj") -> Csproj
-    | f when f.EndsWith("vbproj") -> Vbproj
-    | _                           -> failwith (sprintf "Project file %s not supported. Unknown project type." projFileName)
+let generateAssemblyInfo _ =
 
+    let (|Fsproj|Csproj|Vbproj|) (projFileName:string) =
+        match projFileName with
+        | f when f.EndsWith("fsproj") -> Fsproj
+        | f when f.EndsWith("csproj") -> Csproj
+        | f when f.EndsWith("vbproj") -> Vbproj
+        | _                           -> failwith (sprintf "Project file %s not supported. Unknown project type." projFileName)
 
-Target.create "AssemblyInfo" <| fun _ ->
     let releaseChannel =
-        match release.SemVer.PreRelease with
+        match releaseNotes.SemVer.PreRelease with
         | Some pr -> pr.Name
         | _ -> "release"
     let getAssemblyInfoAttributes projectName =
         [ AssemblyInfo.Title (projectName)
           AssemblyInfo.Product productName
-          AssemblyInfo.Version release.AssemblyVersion
-          AssemblyInfo.Metadata("ReleaseDate", release.Date.Value.ToString("o"))
-          AssemblyInfo.FileVersion release.AssemblyVersion
-          AssemblyInfo.InformationalVersion release.AssemblyVersion
+          AssemblyInfo.Version releaseNotes.AssemblyVersion
```

Comment:
more and more I wonder about this assemblyinfo file generation.  The SDK seems to want us to supply these things via msbuild props, maybe it's worth writing to a `Directory.Build.props` instead or adding as msbuild properties?

Reply:
I remember there being some issues with how the AssemblyVersion was generated with the F# SDK.  Like it would spit out `0.0.0.0` if you asked for it via reflection.  Has the been fixed in a reliable manner?

Reply:
https://github.com/dotnet/fsharp/issues/3113#issuecomment-494088545

Reply:
looks like it's still unresolved

Reply:
![sadkitty](https://user-images.githubusercontent.com/1490044/58042752-7d61e300-7b09-11e9-8fc8-5a3a56cceffd.jpeg)